file_util/inc/leatherman/file_util/file.hpp

```diff
@@ -0,0 +1,117 @@
+/**
+ * @file
+ * Declares utility functions for reading data from files.
+ */
+#pragma once
+
+#include <boost/filesystem.hpp>
+
+#include <string>
+#include <stdexcept>
+#include <functional>
+#include <wordexp.h>
+#include <fstream>
+
+namespace leatherman { namespace file_util {
+
+    /// Generic file error class.
+    class file_error : public std::runtime_error {
+    public:
+        explicit file_error(std::string const& msg) : std::runtime_error(msg) {}
+    };
+
+    /**
+     * Contains utility functions for reading data from files.
+     */
+    struct __attribute__((visibility("default"))) file
```

Comment:
Same question as with the directory file: why bundled in a struct? Should it stay this way?